app.js

```diff
@@ -60,37 +37,72 @@ let fetch = (user, repo) => {
       sort: "updated",
       order: "desc",
       since: ghCommit.commit.author.date
-    });
-  }).then((issues) => {
-    const json = {};
+    })
+  }
+}
+
+const testLabel = (l) => { return regex.test(l.name) }
+
+const fetchCommits = (user, repo) => {
+  return github.repos.getCommits({
+    user,
+    repo,
+    sha: "production"
+  })
+}
+
+const collectIssues = (issues) => {
+  return new Promise((resolve, reject) => {
+    let json = {}
 
     // iterate through each issue
     _.each(issues, (issue) => {
       // grab the labels we care about
       issue.labels.filter(testLabel).forEach((label) => {
         if (json[label.name] == null) {
-          json[label.name] = [];
+          json[label.name] = []
         }
 
-        json[label.name].push(issue);
-      });
-    });
+        json[label.name].push(issue)
+      })
+    })
+
+    resolve(json)
+  })
+}
 
+const output = (user, repo) => {
```

Comment:
Alternatives for this partial/currying style of functions would be:
```js
const output = _.curry((user, repo, issues) => {
  ...
})
output(user)(repo)(issues)
output(user, repo)(issues)
output(user)(repo, issues)
output(user, repo, issues)
```
or partials
```js
const output = (user, repo, issues) => {
  ...
}
_.partial(output, user, repo)(issues)
```

Where currying makes a function so you can call it with partial arguments and it will return a function with those arguments stored, and it will call the final function once it has all of the arguments available.   And partials are similar except more explicit in that you always have to call `_.partial` to add arguments (it's the one I use more often by far, and I generally prefer the explicit nature for readability and to avoid silent errors)